src/tir/analysis/var_touch.cc

```diff
@@ -45,12 +47,16 @@ class VarTouchVisitor : public StmtExprVisitor {
   void VisitExpr_(const VarNode* op) final { Handle(op); }
 
   void VisitStmt_(const BufferStoreNode* op) final {
-    Handle(op->buffer->data.get());
+    if (touch_write_) {
```

Comment:
This looks like it's a repetition of the logic in `VarUseDefAnalyzer`, which already distinguishes between variable access, reads of a buffer, and writes to a buffer.  I'd recommend using it rather than duplicating similar logic here.